pkg/cmd/extension/command.go

```diff
@@ -293,19 +293,35 @@ func NewCmdExtension(f *cmdutil.Factory) *cobra.Command {
 				Use:   "install <repository>",
 				Short: "Install a gh extension from a repository",
 				Long: heredoc.Docf(`
-					Install a GitHub repository locally as a GitHub CLI extension.
+					Install a GitHub CLI extension from a GitHub repository.
 
-					The repository argument can be specified in %[1]sOWNER/REPO%[1]s format as well as a full URL.
+					The repository argument can be specified in %[1]sOWNER/REPO%[1]s format or as a full repository URL.
 					The URL format is useful when the repository is not hosted on github.com.
 
-					To install an extension in development from the current directory, use %[1]s.%[1]s as the
-					value of the repository argument.
+					To install an in-development extension from a locally cloned repository, use %[1]s.%[1]s as the
+					value of the repository argument. Note the following:
+
+					- After installing an extension from a locally cloned repository, the GitHub CLI will
+					manage this extension as a symbolic link pointing to an executable file with the same
+					name as the repository in the repository's root.
+						- For example, if the repository is named %[1]sgh-foobar%[1]s, the symbolic link will
```

Comment:
From the tests below, it seems like what's described here is the non-windows behavior:

> // We don't create symlinks on Windows, so check if we made a file
		// with the correct contents

How is windows different and should we enumerate those differences here?

Reply:
I find the idea that the test works as documentation of it fairly compelling. Since it's symlink-like for windows, there's probably no need to enumerate the semantics in our docs. Notably, I also didn't pick up on the windows difference until I saw the test, so relying on the test for that documentation seems reasonable, since the answer to what we do is right there in the test. Maybe we update the test comment to be a bit more specific? I'll give a stab at it.

Reply:
Sounds good to me! Thanks for the suggestion ✨ , I committed it.